scripts/service-worker-integration.ts

```diff
@@ -0,0 +1,64 @@
+import type { APIRoute, AstroIntegration } from 'astro';
+import { fileURLToPath } from 'node:url';
+import * as esbuild from 'esbuild';
+
+const serviceWorkerSrc = fileURLToPath(new URL('../src/assets/service-worker.ts', import.meta.url));
```

Comment:
I think within an Astro integration you have access to the configured src and dist directory (`srcDir` and `outDir` on the `config` param) ([reference](https://github.com/withastro/astro/blob/main/packages/integrations/sitemap/src/index.ts#L70-L78)).

So I would change the API of the `serviceWorkerIntegration` so it can be called as something like:

```ts
serviceWorkerIntegration({
  srcFilename: 'assets/service-worker.ts', // relative to `srcDir`,
  outFilename: 'service-worker.ts',        // relative to `outDir`,
})
```

Reply:
We tried. Had issues hoisting to root scope. Slightly refactored instead.